background/adapter/deluge.js

```diff
@@ -30,28 +30,40 @@ torrentToWeb.adapter.deluge = function (baseUrl, username, password, autostart)
     }
 
     return {
-        send: function (filename, data, callback) {
-            let fileReader = new FileReader();
+        send: function (filenameOrUrl, data, callback) {
+            let options = {};
 
-            fileReader.addEventListener('load', function () {
-                let options = {};
-
-                if (! autostart) {
-                    options['add_paused'] = true;
-                }
+            if (! autostart) {
+                options['add_paused'] = true;
+            }
 
+            if (filenameOrUrl.startsWith('magnet:')) {
                 let requestData = {
                     id: '2',
-                    method: 'core.add_torrent_file',
-                    params: [filename, window.btoa(fileReader.result), options],
+                    method: 'core.add_torrent_magnet',
+                    params: [filenameOrUrl, options],
                 };
 
                 login(() => {
                     sendRequest(requestData, callback, null);
                 });
-            });
+            } else {
```

Comment:
No need for an else block here, early return is preferred :) – Same applies to the other adapters.

Reply:
> No need for an else block here, early return is preferred :) – Same applies to the other adapters.

Huh? Read the code again. There is no early return possible here!
Pseudocode:

if (isMagnet) {
  handleMagnet;
} **else** {
  handleTorrent;
}

Reply:
Or do you want an explicit return after handleMagnet?